src/main/fc/rc_modes.c

```diff
@@ -72,29 +71,44 @@ bool isRangeActive(uint8_t auxChannelIndex, const channelRange_t *range) {
 
 void updateActivatedModes(void)
 {
-    boxBitmask_t newMask;
+    boxBitmask_t newMask, andMask;
     memset(&newMask, 0, sizeof(newMask));
-
-    for (int index = 0; index < MAX_MODE_ACTIVATION_CONDITION_COUNT; index++) {
-        const modeActivationCondition_t *modeActivationCondition = modeActivationConditions(index);
-
-        if (isRangeActive(modeActivationCondition->auxChannelIndex, &modeActivationCondition->range)) {
-            boxId_e mode = modeActivationCondition->modeId;
-            if (mode < CHECKBOX_ITEM_COUNT)
-                bitArraySet(&newMask, mode);
+    memset(&andMask, 0, sizeof(andMask));
+
+    // determine which conditions set/clear the mode
+    for (int i = 0; i < MAX_MODE_ACTIVATION_CONDITION_COUNT; i++) {
+        const modeActivationCondition_t *mac = modeActivationConditions(i);
+
+        boxId_e mode = mac->modeId;
+        if (mode < CHECKBOX_ITEM_COUNT) {
+            bool bAnd = mac->modeLogic || bitArrayGet(&andMask, mode);
+            bool bAct = isRangeActive(mac->auxChannelIndex, &mac->range);
+            if (bAnd) {
+                bitArraySet(&andMask, mode); // used later for XORing with newMask
+                if (!bAct)
```

Comment:
This will be confusing when first first conditions are not marked - the behave as if negated